### PR TITLE
Add standard Ethereum precompiles and re-generate Algol spec

### PR DESCRIPTION
**Pull request by branan**
# Description

This adds the standard EVM precompiles (necessary for Axelar and for running smart contracts in general), and updates the Algol spec to the latest version of the code.

Since Algol never really went successfully  live, re-generating the spec is fine.

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by mustermeiszer** [comment]
@NunoAlexandre can you take over this one and add the needed pallets to Altair?

File: runtime/common/src/evm/precompile.rs [hunk 1]
```diff
@@ -15,8 +15,33 @@ use core::marker::PhantomData;
...
+const BN128PAIRING_ADDR: Addr = addr(8);
+const BLAKE2F_ADDR: Addr = addr(9);
+// 1024->2047: Nonstandard precompiles shared with other chains (such
+// as Moonbeam). See
+// https://docs.moonbeam.network/builders/pallets-precompiles/precompiles/overview/#precompiled-contract-addresses
+const SHA3FIPS256_ADDR: Addr = addr(1024);
+const DISPATCH_ADDR: Addr = addr(1025);
+const ECRECOVERPUBLICKEY_ADDR: Addr = addr(1026);
+// 2048-XXXX: Nonstandard precompiles that are specific to our chain.
+// This section intentionally left blank
```
Comment (NunoAlexandre) [inline runtime/common/src/evm/precompile.rs:43]: 👌

**Review by mustermeiszer — APPROVED** [review]
LGTM!

File: runtime/common/src/evm/precompile.rs [hunk 2]
```diff
@@ -26,41 +51,57 @@ impl<R> CentrifugePrecompiles<R> {
...
+			MODEXP_ADDR,
+			BN128ADD_ADDR,
+			BN128MUL_ADDR,
+			BN128PAIRING_ADDR,
+			BLAKE2F_ADDR,
+			SHA3FIPS256_ADDR,
+			DISPATCH_ADDR,
+			ECRECOVERPUBLICKEY_ADDR,
+		]
+		.contains(&address.0)
```
Comment (NunoAlexandre) [inline runtime/common/src/evm/precompile.rs:93]: Performance doesn't really matter with such a small sample but, as a nit, we could make this a `match` statement for improved performance and scalability:



```suggestion
		match &address.0 {
			  ECRECOVER_ADDR |
			  SHA256_ADDR |
			  RIPEMD160_ADDR |
			  IDENTITY_ADDR |
			  MODEXP_ADDR |
			  BN128ADD_ADDR |
			  BN128MUL_ADDR |
			  BN128PAIRING_ADDR |
			  BLAKE2F_ADDR |
			  SHA3FIPS256_ADDR |
			  DISPATCH_ADDR |
			  ECRECOVERPUBLICKEY_ADDR => true,
			  _ => false,
		}
```

Comment (NunoAlexandre) [reply runtime/common/src/evm/precompile.rs:93]: or just use `matches!(`

**Review by NunoAlexandre — APPROVED** [review]
🚆  let's go! left some nits but not blockers by any measure, happy to merge this as is

**Pull request merged**